commit 7f6f5a06a070
Author: Nils Ohlmeier [:drno] <drno@ohlmeier.org>
Date:   Wed May 16 09:51:35 2018 -0700

    Bug 1439236 - Exit early if m_copym gets called with null ptr. r=dminor, a=RyanVM
    
    Cheery-picked upstream commit:
    https://github.com/sctplab/usrsctp/commit/d89882d04900c80860874b5eb389b3ed3a0bca3d
    
    MozReview-Commit-ID: 36bYbfIaqEz
    
    --HG--
    extra : source : 2755492fe3d1daa9dd6066b049bc7bb5941b28e9
---
 netwerk/sctp/src/user_mbuf.c | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git netwerk/sctp/src/user_mbuf.c netwerk/sctp/src/user_mbuf.c
index ec83931e66e5a..bd17f9fbada60 100755
--- netwerk/sctp/src/user_mbuf.c
+++ netwerk/sctp/src/user_mbuf.c
@@ -987,6 +987,13 @@ m_copym(struct mbuf *m, int off0, int len, int wait)
 
 	KASSERT(off >= 0, ("m_copym, negative off %d", off));
 	KASSERT(len >= 0, ("m_copym, negative len %d", len));
+	KASSERT(m != NULL, ("m_copym, m is NULL"));
+
+#if !defined(INVARIANTS)
+	if (m == NULL) {
+		return (NULL);
+	}
+#endif
 
 	if (off == 0 && m->m_flags & M_PKTHDR)
 		copyhdr = 1;
